examples/data-objects/todo/README.md

```diff
@@ -18,32 +18,28 @@ You can run this example using the following steps:
 1. Run `npm run start` from this directory (examples/data-objects/todo) and open <http://localhost:8080> in a web browser to see the app running.
 <!-- AUTO-GENERATED-CONTENT:END -->
 
-## Todo Fluid Objects
+## Todo Fluid objects
 
 There are two Fluid objects that make up the Todo application:
 
-### [Todo](./src/Todo/index.tsx)
+### [Todo](./src/Todo/Todo.ts)
 
-A Todo is the top level Fluid object and contains three core concepts:
+A Todo is the top level Fluid object.  It can create, delete, and provides access to TodoItems.  It also has an editable title (using a SharedString).
 
-1. Title
-2. Ability to create new Todo Items
-3. Collection of Todo Items
+### [TodoItem](./src/TodoItem/TodoItem.ts)
 
-### [TodoItem](./src/TodoItem/index.tsx)
+A Todo Item is a singular todo entry.  It has editable text as well as editable detail text (also using a SharedString).  It also stores a boolean for the checkbox.
 
-A Todo Item is a singular todo entry. Because each Todo Item is its own Fluid object each Todo Item can be independently opened.
+## The views
 
-Todo Items can contain one inner Fluid object. These can currently be another Todo Item or a Clicker.
+### [TodoView](./src/Todo/TodoView.tsx)
 
-## Other Fluid Objects
+The default view is a TodoView mapping to a Todo.  It uses the CollaborativeInput control from the `@fluid-experimental/react-inputs` package to display its title, and TodoItemViews to display the data from its TodoItems.  It also provides a text field and button for creating new TodoItems, and buttons for each TodoItem to delete or open directly.
 
-There are two other Fluid objects that live in the todo repo but are there to extend functionality of `TodoItem`
+### [TodoItemView](./src/TodoItem/TodoItemView.tsx)
 
-### [TextBox](./src/TextBox/index.tsx)
+The TodoItemView uses the CollaborativeInput as well as a CollaborativeTextArea to display its string contents, and uses a plain HTML checkbox for the checked state.
 
-A Collaborative TextArea based off the React CollaborativeTextArea. This should be replaced by just pulling in the `@fluid-example/collaborative-textarea` Fluid objects.
+## The container code
 
-### [TextList](./src/TextList/index.tsx)
-
-Used the `SharedString` to manage a list of text items.
+The container code includes a request handler that provides views to the data stored within.  For a default (empty) request, it provides a TodoView in response.  The container code's request handler can also provide back a TodoItemView directly when a direct link is used.
```

Comment:
In these examples we are still providing views through containers and `request` pattern. Since we discussed discouraging usage of `request` pattern, I'm not sure we have substitute functionality in this case? We chatted about this earlier, I'm just curious if we have any new takes on this.

I understand in terms of usability we are getting views for free (at runtime too), but now that we are talking about splitting views and data, more broadly looking, is this really common pattern that (general) user would compose views with? In other words, they would "request" the view from container and compose further with it?
If we use containers as data-only collaborations, and we separate views and presentational concerns - do we lose much?

Reply:
Good question -- as I worked through this, the soft stake in the ground I'm considering is "DataObjects shouldn't do routing", but I don't think we're quite ready yet to say "ContainerRuntimes shouldn't do routing" since we don't have a great alternative entrypoint from the IContainer.

This is why I have the ContainerRuntime request handler route all the way to the TodoItem itself, rather than e.g. routing to the Todo and then delegating to the Todo the task of routing to the specific TodoItem.

I do still recommend the external-views approach (container gives data only) as the best option, but for now I think I'd say the container-views approach taken here and with the rest of the webpack-fluid-loader demos is "not recommended, but supported".  We should probably continue exploring good patterns for customers that do this (or similar) things today to help them move in the direction of data-only containers.

Reply:
We should maybe revise and be more explicit who these examples are for. FFX is mature enough and they have their own ideas around usage patterns, examples etc. If this is for "general" FF consumer, I do not clearly see the significant value of composing with container-views. It sort of forces you to rethink how you approach view composition, sharing data, decupling view/data etc. If there is no clear "we want you to get into these advanced usages because they are amazing at X, and solving X is a common-enough problem", we could add this to our "get rid of" list (I'd  be curious to learn from FFX experience in this regard, since I think they are relying on this pattern quite a bit - but I could be wrong).

All of this is general comment, your PR is taking many steps in the right direction. With these simplifications you made, it makes it sort of easier to reason with and see some of the remaining gaps. Nice work!

Btw, do we have an epic/story calling out patterns we want to promote/remove ?

Reply:
All very true - I think there's a discussion to be had there.  The purported benefit of the container-views approach is for supporting hosts who want to load and use a container that they did not write themselves, and treat it as a self-contained black box of data+views -- as FFX intends to support today with their office container.  There are other valid approaches to this problem that we could consider though.

Part of what you see here is an evolution of the framework -- originally the container-views approach was the recommendation, and in fact was the only pattern we demonstrated.  Over time we have established and shifted preference to data-only containers, but I think there's more due diligence required to fully write off the container-views approach.

I'm tracking removal of IFluidHTMLView in general via #1041.  Spaces is the last demo using the innerRequestHandler pattern, which I'm hoping to do a rewrite of soon as well.